examples/nextjs-flow/app/hooks/useTranscriptManager.ts

```diff
@@ -0,0 +1,34 @@
+import { useFlowEventListener } from '@speechmatics/flow-client-react';
+import { useState, useRef, useEffect } from 'react';
+import { type TranscriptGroup, TranscriptManager } from './newPartials';
+
+// Custom hook to manage TranscriptManager instance
+export function useTranscriptManager() {
+  const [transcriptGroups, setTranscriptGroups] = useState<TranscriptGroup[]>(
+    [],
+  );
+  const managerRef = useRef<TranscriptManager | null>(null);
+
+  useEffect(() => {
+    const manager = new TranscriptManager();
```

Comment:
The way you've written the `TranscriptManager` class actually sets it up perfectly for [`useSyncExternalStore`](https://react.dev/reference/react/useSyncExternalStore)

Rather than the `useEffect` setup, you can define two functions, get (which returns the transcript groups) and subscribe (which notifies on change), and that hook will take care of the rest :)

Another benefit is it means you don't need to keep `transcriptGroups` in React state at all

Reply:
Isn't that still using react state though?

Reply:
Possibly, I guess I just meant we don't need a `useState` if we use that

Reply:
Note - this is to be refined at a later time